rusk-wallet/src/lib/prompt.rs

```diff
@@ -372,21 +361,25 @@ fn is_valid_addr(addr: &str) -> bool {
     bs58::decode(addr).into_vec().is_ok()
 }
 
-fn check_valid_denom(num: Dusk, _prev: &Answers) -> Result<(), String> {
+fn check_valid_denom(num: Dusk, balance: Dusk) -> Result<(), String> {
     if num.is_finite() && num > 0.0 {
-        Ok(())
+        if num < balance {
+            Ok(())
+        } else {
+            Err("insufficient balance".to_owned())
+        }
     } else {
         Err("invalid denomination".to_owned())
     }
 }
```

Comment:
Instead, I think it could make more sense to perform a local check with the total cost for the tx right before sending it through wallet core, and spitting an error if there's not enough balance.

Reply:
Rearranging the options wouldn't make sense from the user's perspective though. Lets leave it like this for now, and make it better later.